hazelcast/src/main/java/com/hazelcast/jet/impl/client/protocol/task/JetTerminateJobMessageTask.java

```diff
@@ -46,4 +54,17 @@ public String getMethodName() {
     public Object[] getParameters() {
         return new Object[]{};
     }
+
+    @Nullable
+    @Override
+    public String[] actions() {
+        TerminationMode terminationMode = TerminationMode.values()[parameters.terminateMode];
+        if (terminationMode == RESTART_GRACEFUL || terminationMode == RESTART_FORCEFUL) {
+            return new String[]{ActionConstants.ACTION_CREATE, ActionConstants.ACTION_DESTROY};
+        }
+        if (terminationMode == SUSPEND_GRACEFUL || terminationMode == SUSPEND_FORCEFUL) {
+            return new String[]{ActionConstants.ACTION_SUSPEND};
+        }
+        return new String[]{ActionConstants.ACTION_DESTROY};
```

Comment:
I also changed CANCEL+DESTROY to SUSPEND+RESUME.
```suggestion
        ActionAfterTerminate terminationMode = TerminationMode.values()[parameters.terminateMode].actionAfterTerminate();
        switch (terminationMode) {
            case RESTART:
                return new String[]{ActionConstants.ACTION_SUSPEND, ActionConstants.ACTION_RESUME};

            case SUSPEND:
                return new String[]{ActionConstants.ACTION_SUSPEND};

            case CANCEL:
                return new String[]{ActionConstants.ACTION_DESTROY};
        }
        
        throw new RuntimeException(terminationMode.toString());
```